### ISSUE TITLE
VS Code: Launch Playground in VS Code using "blueprint mode"

### ISSUE THREAD

**Issue by flexseth** [WordPress/playground-tools#211]
When launching WordPress in VS Code via the extension, search first for a `blueprint.json` file to bootstrap the instance. 

If a `blueprint.json` file is found in the project directory, load up the blueprint at the click of a button. 

<img width="538" alt="image" src="https://github.com/WordPress/wordpress-playground/assets/3792502/0a6a9d64-347a-4404-bad2-9179d05f1168">

## Current available modes
> The WordPress Playground VS Code Extension [automatically operates ](https://marketplace.visualstudio.com/items?itemName=WordPressPlayground.wordpress-playground#:~:text=currently%20have%20open.-,Automatic%20Modes,-The%20WordPress%20Playground)in a few different modes. The selected mode depends on the project directory in which it is executed

### Automattic modes
- plugin, theme, or wp-content
- wordpress
- wordpress-develop
- index
- playground

**Comment by adamziel** [comment]
+1 on that, or some form of it that allows for multiple Blueprints and is consistent with how Blueprints are supported in GitHub projects.

I'll move this over to the https://github.com/WordPress/playground-tools/ repo where the VS Code extension is maintained.

cc @sejas

**Comment by sejas** [comment]
I love the idea! The only thing to bear in mind is that the blueprint should be executed only the first time, because blueprints are expected to create a "running" WordPress instance once.

**Comment by flexseth** [comment]
> the blueprint should be executed only the first time

great phrasing @sejas - adding this to the docs. It's an important distinction to make

**Comment by flexseth** [comment]
> The only thing to bear in mind is that the blueprint should be executed only the first time, because blueprints are expected to create a "running" WordPress instance once.

@jonathanbossenger did a [livestream ](https://wordpress.tv/2024/04/04/testing-wp-playground-with-wp-now/) on the Playground, he was able to get a new site generated by running the blueprint again - using `wp-now`

In his example, he changed the blueprint step to change the site name. 
Around 1 hour into the livestream.

If a blueprint can re-write an entire site, it probably would be helpful to throw some type of **Caution** message to let the user know the entire site could be re-written.

Following progress - feel like this has opportunity to be a great feature.
One-click from VS Code or command-line to a WordPress demo website.

**Comment by sejas** [comment]
Yes! Some blueprints are less destructive.
Maybe we shouldn't autorun a blueprint by default, and instead the user can run it on demand with one click from the UI, as many times as they want.
My concern is that maybe some blueprints could be used to hack/or inject some malicious code in the user's computer.

**Comment by flexseth** [comment]
> Maybe we shouldn't autorun a blueprint by default, and instead the user can run it on demand with one click from the UI, as many times as they want.

Absolutely. The UI for VS Code I think could offer a ton more options. While using the extension I'll try to track down a "Wishlist" of sorts for things I'd love to see. It's the main way I could see using Playground!! 

> My concern is that maybe some blueprints could be used to hack/or inject some malicious code in the user's computer.
From my understanding, due to the "containerized" implementation of the Playground... this is a limited concern?

On the other hand I could see DDoS attacks or other outgoing attacks being more prevelant?

Disclaimer: Not a security expert by any way, but this is a feature that is appealing of Playground (less danger)

**Comment by flexseth** [comment]
> Maybe we shouldn't autorun a blueprint by default, and instead the user can run it on demand with one click from the UI, as many times as they want.

I was working on generating some use journeys that could create errors. GitHub Copilot is magical, by the way... nonetheless, here's two user scenerios it generated for blueprint errors: 

### User Journey 1: User is trying to import a Blueprint and gets an error

1. User navigates to the Playground dashboard
2. User clicks on the "Blueprints" tab
3. User clicks on the "Import Blueprint" button
4. User selects a Blueprint file to import
5. User clicks the "Import" button
6. User receives an error message

### User Journey 2: User is trying to create a new Blueprint and gets an error

1. User navigates to the Playground dashboard
2. User clicks on the "Blueprints" tab
3. User clicks on the "Create Blueprint" button
4. User fills out the Blueprint form
5. User clicks the "Create" button
6. User receives an error message

---

Which could actually provide some ideas for how the UI could work!
